eduNEXT/xblock-filesmanager
# fix: removing delete action for files and adding delete folders only

### Description

This allows deleting only folders in the files manager tree, except for the 'unpublish' folder. If a folder contains assets and is deleted, those assets will be moved to the 'unpublish' folder.

![filesmanager-1](https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/1e9673f1-c045-47e3-b1b7-e95aab33fbb2)

##### Asset after deleting a folder with files inside
![filesmanager-2](https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/8be8d2d6-8fb8-4dff-98e3-144b242279cb)


### How to test it

- Check this branch`jv/fix-remove-delete-action` in the XBlock files manager
- Create a new branch 

 ```
git checkout -b branch-to-test-integration
```
- Pull the changes made in [this pr](https://github.com/eduNEXT/xblock-filesmanager/pull/21)

 ```
git pull origin MJG/avoid-casting-filesize
```
1. use node 18
 ```
nvm use
```

2. install dependencies
```
yarn install 
```
3 run to see any change in the react code 
```
yarn dev 
```
------------------------------------------------------------------------------------------------------------------------------------------------
Now select the folders to delete (Ctrl + click) -> "Delete folders"

![delete-folders](https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/79b53c2b-bd2b-4def-94d7-1add075d9c2e)

If you have files selected the option "Delete folders" shouldn't appear

![delete-folder-no-option](https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/94f0d38d-7a57-4255-843e-dec23eccfcfe)


You can not delete "Unpublish" folder
![unpusblish-folder](https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/f1b30adc-ae31-4323-9764-c8f7b5f6def5)




**Merge checklist:**
Check off if complete *or* not applicable:
